### PR TITLE
removed grid processing from pose_class_disentanglement_plot

### PR DESCRIPTION
**Pull request by marjanfamili**
Addresses #260 
- [x]  remove batch process from `plot_classes_distribution`
- [x] remove batch process from `interpolations_plot`
- [x] remove batch process from `latent_disentamglement_plot`
- [x] remove batch process from `latent_4enc_interpolate_plot`
- [x] remove batch process from `pose_class_disentanglement_plot`

### CODE REVIEW

File: avae/evaluate.py [hunk 1]
```diff
@@ -169,17 +169,30 @@ def evaluate(
     # visualise latent disentanglement
     if settings.VIS_DIS:
         vis.latent_disentamglement_plot(
-            x_test, vae, device, data_dim, poses=p_test, mode="_eval"
+            batch[0].shape[-data_dim:],
```
Comment (crangelsmith) [inline avae/evaluate.py:172]: it took me a bit to understand what was this for, could we define a variable size once (dsize?) and then reuse in all the different function. For readability.

File: avae/evaluate.py [hunk 2]
```diff
@@ -169,17 +169,30 @@ def evaluate(
...
+            x_test,
+            vae,
+            device,
+            data_dim,
```
Comment (crangelsmith) [inline avae/evaluate.py:176]: would dsize and data_dim redundant?

File: avae/utils.py [hunk 3]
```diff
@@ -228,3 +230,29 @@ def colour_per_class(classes: list):
     # Generate a list of colors based on the modulo operation of i with respect to the number of colors in the combined colormap
     colours = [custom_cmap(i % len(combined_cmap)) for i in range(num_colors)]
     return colours
+
+
+def pose_interpolation(
```
Comment (crangelsmith) [inline avae/utils.py:235]: Can we add some documentation to this function? As is a complex one and has many different variables.

**Comment by crangelsmith** [comment]
I've added some suggestions, as this PR not going to develop yet I'm ok to merge it and implement it in the resulting branch.

File: avae/evaluate.py (hunk 2 above)
Comment (marjanfamili) [reply avae/evaluate.py:177]: no no, it is the data dimension so a list of for example [64,64,64] , you comment is a good suggestion , implementing now

File: avae/evaluate.py (hunk 1 above)
Comment (marjanfamili) [reply avae/evaluate.py]: This is done now

**Review by crangelsmith — APPROVED** [review]

File: avae/utils.py (hunk 3 above)
Comment (marjanfamili) [reply avae/utils.py:235]: This is done now

**Pull request merged**